### PR TITLE
[monitoring-docs] Restructured Documentation: New Unified Documentation Website

### PR DESCRIPTION
**Pull request by pandafy**
Related to [openwisp/openwisp-docs#107](https://github.com/openwisp/openwisp-docs/issues/107).
These documents are pulled by the logic being worked on in [openwisp/openwisp-docs#189](https://github.com/openwisp/openwisp-docs/pull/189).
We are doing similar changes in all modules.

### CODE REVIEW

File: docs/developer/utils.rst [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+failed write task will be retried in the background so that there is no
+loss of data if failures occur due to overload of Timeseries server. You
+can read more about this retry mechanism at
+`OPENWISP_MONITORING_WRITE_RETRY_OPTIONS
+<#openwisp-monitoring-write-retry-options>`_.
+
+``InvalidMetricConfigException``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+**Path**:
```
Comment (nemesifier) [inline docs/developer/utils.rst:484]:
```suggestion
**Full python path**:
```
I've used the same in OpenWISP Users. Applies to all occurrences (use search and replace).

File: docs/developer/utils.rst [hunk 2]
```diff
@@ -0,0 +1,495 @@
...
+``openwisp_monitoring.monitoring.exceptions.InvalidMetricConfigException``
+
+This exception shall be raised if the metric configuration is broken.
```
Comment (nemesifier) [inline docs/developer/utils.rst:487]:
```suggestion
This exception will be raised if the metric configuration is broken.
```

File: docs/developer/utils.rst [hunk 3]
```diff
@@ -0,0 +1,495 @@
...
+
+``InvalidChartConfigException``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+**Path**:
+``openwisp_monitoring.monitoring.exceptions.InvalidChartConfigException``
+
+This exception shall be raised if the chart configuration is broken.
```
Comment (nemesifier) [inline docs/developer/utils.rst:495]:
```suggestion
This exception will be raised if the chart configuration is broken.
```

File: docs/developer/utils.rst [hunk 4]
```diff
@@ -0,0 +1,495 @@
...
+**Path**: ``openwisp_monitoring.db.exceptions.TimeseriesWriteException``
+
+If there is any failure due while writing data in timeseries database,
+this exception shall be raised with a helpful error message explaining the
+cause of the failure. This exception will normally be caught and the
+failed write task will be retried in the background so that there is no
+loss of data if failures occur due to overload of Timeseries server. You
+can read more about this retry mechanism at
+`OPENWISP_MONITORING_WRITE_RETRY_OPTIONS
+<#openwisp-monitoring-write-retry-options>`_.
```
Comment (nemesifier) [inline docs/developer/utils.rst:479]: Please ensure links are fixed one by one (do a project wide search and look for `<#`).

File: docs/user/intro.rst [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  :ref:`jitter <iperf3>`, :ref:`datagram <iperf3>`, :ref:`datagram loss
+  <iperf3>`
+- Maintains a record of :doc:`WiFi sessions <wifi-sessions>` with clients'
+  MAC address and vendor, session start and stop time and connected device
+  along with other information
+- Charts can be viewed at resolutions of the last 1 day, 3 days, 7 days,
+  30 days, and 365 days
+- Configurable alerts
+- CSV Export of monitoring data
+- An overview of the status of the network is shown in the admin
```
Comment (nemesifier) [inline docs/user/intro.rst:31]: let's move this to the top as one of the first items

File: docs/user/intro.rst [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  offline or having issues; there are also :doc:`two timeseries charts
+  which show the total unique WiFI clients and the traffic flowing to the
+  network <dashboard-monitoring-charts>`, a geographic map is also
+  available for those who use the geographic features of OpenWISP
+- Possibility to configure additional :ref:`Metrics
+  <openwisp_monitoring_metrics>` and :ref:`Charts
+  <openwisp_monitoring_charts>`
+- Extensible active check system: it's possible to write additional checks
+  that are run periodically using python classes
+- Extensible metrics and charts: it's possible to define new metrics and
```
Comment (nemesifier) [inline docs/user/intro.rst:42]: let's link the relevant section here (eg: the setting)

File: docs/user/quickstart.rst [hunk 7]
```diff
@@ -0,0 +1,103 @@
+Quickstart Guide
+================
+
+Install monitoring packages on the device
```
Comment (nemesifier) [inline docs/user/quickstart.rst:4]: let's capitalize all headings as indicated in other comments

File: docs/user/rest-api.rst [hunk 8]
```diff
@@ -0,0 +1,282 @@
...
+<https://www.django-rest-framework.org/topics/browsable-api/>`_, which
+makes it even easier to find out the details of each endpoint.
+
+.. _monitoring_rest_endpoints:
+
+List of endpoints
+-----------------
+
+Since the detailed explanation is contained in the :ref:`Live
+documentation <monitoring_live_documentation>` and in the :ref:`Browsable
```
Comment (nemesifier) [inline docs/user/rest-api.rst:38]: in cases like here, where the target heading has the same title of the link text, we can just shorten this, eg: `:ref:`<monitoring_live_documentation>`. This avoids some redundancy.

File: docs/user/metrics.rst [hunk 9]
```diff
@@ -0,0 +1,299 @@
...
+The the different :doc:`device metric <./metrics>` collected by OpenWISP
+Monitoring can be divided in two categories:
+
+1. **metrics collected actively by OpenWISP**: these metrics are collected
+   by the celery workers running on the OpenWISP server, which
+   continuously sends network requests to the devices and store the
+   results;
+2. **metrics collected passively by OpenWISP**: these metrics are sent by
+   the `openwrt-openwisp-monitoring agent
+   <#install-monitoring-packages-on-the-device>`_ installed on the network
```
Comment (nemesifier) [inline docs/user/metrics.rst:295]: :eyes:

**Review by nemesifier — CHANGES_REQUESTED** [review]
Building now from scratch raises the following warnings for me:

```
/developer/utils.rst:415: WARNING: undefined label: 'checks'
/user/checks.rst:57: WARNING: undefined label: 'adding checks and alert settings from the device page'
/user/intro.rst:6: WARNING: unknown document: 'dashboard_monitoring_charts'
```

Please double check.

File: docs/index.rst [hunk 10]
```diff
@@ -0,0 +1,33 @@
+Monitoring
+==========
+
+OpenWISP Monitoring is a network monitoring system written in Python and
+Django, designed to be **extensible**, **programmable**, **scalable** and
+easy to use by end users: once the system is configured, monitoring
+checks, alerts and metric collection happens automatically.
```
Comment (pandafy) [inline docs/index.rst:7]: I made Google Gemini write this. This is placeholder for now.

Comment (nemesifier) [reply docs/index.rst:7]: What about:

The OpenWISP Monitoring module leverages the capabilities of Python and the Django Framework to provide OpenWISP with robust network monitoring features. Designed to be extensible, programmable, scalable, and user-friendly, this module automates monitoring checks, alerts, and metric collection, ensuring efficient and comprehensive network management.

**Review by nemesifier — CHANGES_REQUESTED** [review]

**Review by nemesifier — CHANGES_REQUESTED** [review]
@pandafy please integrate these changes: https://github.com/openwisp/openwisp-monitoring/commit/fb2814b6dd8213ddb45a8497cf41dfb10eee04a2.

File: README.rst [hunk 11]
```diff
@@ -50,23 +50,23 @@ automation solutions can be built on top of its building blocks.
 Other popular building blocks that are part of the OpenWISP ecosystem are:
 
 - `openwisp-controller
-  <https://github.com/openwisp/openwisp-controller>`_: network and WiFi
+  <https://openwisp.io/docs/__new__/dev/controller/>`_: network and WiFi
```
Comment (nemesifier) [inline README.rst]: Please set the URL without the `__new__` prefix so once we're ready we can just merge it without further editing.

**Review by nemesifier — CHANGES_REQUESTED** [review]

**Review by nemesifier — APPROVED** [review]
Ready

**Pull request merged**